src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -328,13 +333,29 @@ public function url($path)
      *
      * @param  \League\Flysystem\AwsS3v3\AwsS3Adapter  $adapter
      * @param  string  $path
+     * @param  int|string|\DateTime $expires The time at which the URL should
+     *                                      expire. This can be a Unix
+     *                                      timestamp, a PHP DateTime object,
+     *                                      or a string that can be evaluated
+     *                                      by strtotime().
      * @return string
      */
-    protected function getAwsUrl($adapter, $path)
+    protected function getAwsUrl($adapter, $path, $expires)
```

Comment:
That's why I submitted this feature to 5.5 branch, I submitted it yesterday to 5.4 but it was closed as a breaking-change and I was asked to make a pull request to branch 5.5.
also, we discussed this case and no need to but null
[#20151](https://github.com/laravel/framework/pull/20151)